qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,535 @@
+                Note that, while there is no limit in the size of the target that can be
+                constructed, fake backends run on local noisy simulators, and these might
+                present limitations in the number of qubits that can be simulated.
+
+            basis_gates: List of basis gate names to be supported by
+                the target. These must be part of the standard qiskit circuit library.
+                The default set of basis gates is ``["id", "rz", "sx", "x", "cx"]``
+                The ``"reset"``,  ``"delay"``, and ``"measure"`` instructions are
+                always supported by default, even if not specified via ``basis_gates``.
+
+            coupling_map: Optional coupling map
+                for the fake backend. Multiple formats are supported:
+
+                #. :class:`~.CouplingMap` instance
+                #. List, must be given as an adjacency matrix, where each entry
+                   specifies all directed two-qubit interactions supported by the backend,
+                   e.g: ``[[0, 1], [0, 3], [1, 2], [1, 5], [2, 5], [4, 1], [5, 3]]``
+
+                If ``coupling_map`` is specified, it must match the number of qubits
+                specified in ``num_qubits``. If ``coupling_map`` is not specified,
+                a fully connected coupling map will be generated with ``num_qubits``
+                qubits.
+
+            control_flow: Flag to enable control flow directives on the target
+                (defaults to False).
+
+            calibrate_instructions: Instruction calibration settings, this argument
+                supports both boolean and :class:`.InstructionScheduleMap` as
+                input types, and is ``None`` by default:
+
+                #. If ``calibrate_instructions==None``, no calibrations will be added to the target.
+                #. If ``calibrate_instructions==True``, all gates will be calibrated for all
+                    qubits using the default pulse schedules generated internally.
+                #. If ``calibrate_instructions==False``, all gates will be "calibrated" for
+                    all qubits with an empty pulse schedule.
+                #. If an :class:`.InstructionScheduleMap` instance is given, this calibrations
+                    present in this instruction schedule map will be appended to the target
+                    instead of the default pulse schedules (this allows for custom calibrations).
+
+            dtm: System time resolution of output signals in nanoseconds.
+                None by default.
+
+            seed: Optional seed for generation of default values.
+        """
+
+        super().__init__(
+            provider=None,
+            name=f"fake_generic_{num_qubits}q",
+            description=f"This is a fake device with {num_qubits} qubits and generic settings.",
+            backend_version="",
+        )
+
+        self.sim = None
```

Comment:
Can we make the attribute private?

Reply:
Done in [54496db](https://github.com/Qiskit/qiskit/pull/10266/commits/54496db3127ac6a754004a5f77d55770375ad016)